### PR TITLE
chore(trainer): document options usage with RuntimePatch

### LINKED ISSUE #388: Document how to use `options` with TrainerClient
We should update documentation to explain how to use options while creating TrainJob to configure RuntimePatch API.

https://sdk.kubeflow.org/en/latest/

cc @kubeflow/kubeflow-sdk-team @Fiona-Waters 

/good-first-issue
/area docs

### PR DESCRIPTION
**Pull request by nthbotast**
## Summary
- add a Quickstart section that shows how to pass `options` to `TrainerClient.train`
- include a concrete `RuntimePatch` example for setting `service_account_name` through runtime patches
- show combining `RuntimePatch` with a simple metadata option (`Labels`)

## Validation
- `git diff --check`
- lightweight assertion script to confirm the new section and `RuntimePatch` example are present

Fixes #388

### CODE REVIEW

**Review by andreyvelich — COMMENTED** [review]
Thanks for this!

/assign @Fiona-Waters @briangallagher @kramaranya @akshaychitneni @szaher @astefanutti @abhijeet-dhumal 
Please help with review!

File: docs/source/getting-started/quickstart.rst [hunk 1]
```diff
@@ -81,6 +81,62 @@ Use the :class:`~kubeflow.trainer.TrainerClient` to submit your function to Kube
...
+       JobSpecPatch,
+       PodTemplatePatch,
+       PodSpecPatch,
+   )
+   from kubeflow.trainer.types import CustomTrainer
+
+   client = TrainerClient()
+
+   job_name = client.train(
+       trainer=CustomTrainer(func=train_mnist),
```
Comment (Fiona-Waters) [inline docs/source/getting-started/quickstart.rst:110]: This should align with step 2
```suggestion
       trainer=CustomTrainer(func=train_func),
```

File: docs/source/getting-started/quickstart.rst [hunk 2]
```diff
@@ -81,6 +81,62 @@ Use the :class:`~kubeflow.trainer.TrainerClient` to submit your function to Kube
 
 That's it! Your training function is now running on Kubernetes.
 
+Optional: Configure TrainJob options
```
Comment (Fiona-Waters) [inline docs/source/getting-started/quickstart.rst:84]: I think maybe we could move this after step 3, as a standalone?

**Review by Fiona-Waters — COMMENTED** [review]
Thanks for this @nthbotast. Left a couple of comments. 
Please sign off your commit, and check the failing PR checks.

**Comment by andreyvelich** [comment]
@nthbotast Are you still working on this PR, or we should find new contributors to finish it?

**Comment by andreyvelich** [comment]
Resolved in: https://github.com/kubeflow/sdk/pull/722
/close
